Firestore/src/DocumentReference.php

```diff
@@ -387,21 +380,37 @@ public function collection($collectionId)
      * @throws \InvalidArgumentException if an invalid `$options.readTime` is
      *     specified.
      */
-    public function collections(array $options = [])
+    public function collections(array $options = []): ItemIterator
     {
         $options = $this->formatReadTimeOption($options);
 
+        $listCollectionCall = function (array $options) {
+            /**
+             * @var ListCollectionIdsRequest $request
+             * @var CallOptions $callOptions
+             */
+            [$request, $callOptions] = $this->validateOptions(
+                $options,
+                new ListCollectionIdsRequest(),
+                CallOptions::class
+            );
+
+            $response = $this->gapicClient->listCollectionIds($request, $callOptions);
+
+            return $this->serializer->encodeMessage($response->getPage()->getResponseObject());
+        };
```

Comment:
I think it would make more sense to split out the creation of the request and call options from the callable, e.g.

```php
/**
 * @var ListCollectionIdsRequest $request
 * @var CallOptions $callOptions
 */
[$request, $callOptions] = $this->validateOptions(
    $options + ['parent' => $this->name],
    new ListCollectionIdsRequest(),
    CallOptions::class
);

$listCollectionCall = function (array $callOptions) use ($request) {
    $response = $this->gapicClient->listCollectionIds($request, $callOptions);
    return $this->serializer->encodeMessage($response->getPage()->getResponseObject());        
};
```

Then pass `$callOptions` instead of `$options` to the `PageIterator`